src/containers/Transaction/Transaction.js

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useForm, Controller } from 'react-hook-form';
+import * as Component from '@material-ui/core';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import styles from './transaction.module.scss';
+import Text from '../../components/Text';
+
+const defaultValues = {
+  isBuying: true,
+  currencyName: '',
+  price: '',
+  amount: 0,
+  total: ''
+};
+
+const schema = yup.object().shape({
+  currencyName: yup
+    .string()
+    .required(),
+  amount: yup
+    .number()
+    .positive()
+    .integer()
+    .transform((val) => (Number.isNaN(val) ? undefined : val))
+    .required()
```

Comment:
As we talked it is better to keep validation with its messages in here, passed as an argument here.